rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -1,14 +1,51 @@
+require 'activerecord-import'
+
 module RailsEventStoreActiveRecord
   class EventRepository
     def initialize(adapter: Event)
       @adapter = adapter
     end
     attr_reader :adapter
 
-    def create(event, stream_name)
-      data = event.to_h.merge!(stream: stream_name, event_type: event.class)
-      adapter.create(data)
-      event
+    def append_to_stream(events, stream_name, expected_version)
+      events = [*events]
+      expected_version   = case expected_version
+        when nil
+          raise RubyEventStore::InvalidExpectedVersion
+        when :none
+          -1
+        when :auto
+          eis = EventInStream.where(stream: stream_name).order("position DESC").first
```

Comment:
we should use a repository to access this

Reply:
it should also be in its own private method

Reply:
The code we are writing here is the repository :)

Reply:
it's the event repository, not EventInStream repository. I would make a clear distinction